ygot/diff_test.go

```diff
@@ -503,3 +487,553 @@ func TestFindSetLeaves(t *testing.T) {
+		want: false,
+	}, {
+		desc: "multiple paths, equal",
+		inA: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+				}},
+			}},
+		},
+		inB: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+				}},
+			}},
+		},
+		want: true,
+	}, {
+		desc: "multiple paths, unequal",
+		inA: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+				}},
+			}},
+		},
+		inB: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "baz",
+				}},
+			}},
+		},
+		want: false,
+	}, {
+		desc: "multiple paths with keys, equal",
+		inA: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+					Key:  map[string]string{"baz": "bop"},
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+					Key:  map[string]string{"fish": "chips"},
+				}},
+			}},
+		},
+		inB: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+					Key:  map[string]string{"baz": "bop"},
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+					Key:  map[string]string{"fish": "chips"},
+				}},
+			}},
+		},
+		want: true,
+	}, {
+		desc: "multiple paths with keys, equal",
+		inA: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+					Key:  map[string]string{"baz": "bop"},
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+					Key:  map[string]string{"fish": "chips"},
+				}},
+			}},
+		},
+		inB: &pathSpec{
+			gNMIPaths: []*gnmipb.Path{{
+				Elem: []*gnmipb.PathElem{{
+					Name: "foo",
+					Key:  map[string]string{"baz": "bop"},
+				}},
+			}, {
+				Elem: []*gnmipb.PathElem{{
+					Name: "bar",
+					Key:  map[string]string{"fish": "hat"},
+				}},
+			}},
+		},
+		want: false,
+	}, {
+		desc: "both nil",
+		inA:  nil,
+		inB:  nil,
+		want: true,
+	}, {
+		desc: "compare nil",
+		inA:  &pathSpec{},
+		inB:  nil,
+		want: false,
+	}}
+
+	for _, tt := range tests {
+		if got, want := tt.inA.Equal(tt.inB), tt.want; got != want {
+			t.Errorf("%s: (%#v).Equal(%#v): did not get expected result, got: %v, want: %v", tt.desc, tt.inA, tt.inB, got, want)
+		}
+	}
+}
+
+type badGoStruct struct {
+	InvalidEnum int64 `path:"an-enum"`
+}
+
+func (*badGoStruct) IsYANGGoStruct() {}
+
+func TestDiff(t *testing.T) {
+	tests := []struct {
+		desc          string
+		inOrig, inMod GoStruct
+		want          *gnmipb.Notification
+		wantErr       *string
+	}{{
+		desc:   "single path addition in modified",
+		inOrig: &renderExample{},
+		inMod: &renderExample{
+			Str: String("cabernet-sauvignon"),
```

Comment:
I like your selec-erm-collection of testdata ;)

Reply:
;-)